src/components/MovesSection/MovesSection.tsx

```diff
@@ -6,21 +6,23 @@ import useFilterMoves from "./useFilterMoves";
 import { MoveCategory } from "./MoveCategory";
 import { MoveDialog } from "./MoveDialog";
 import SearchIcon from "@mui/icons-material/Search";
+import { useListenToCampaignCustomMoves } from "api/campaign/settings/moves/listenToCampaignCustomMoves";
 
 export interface MovesSectionProps {
   stats?: {
     health: number;
     spirit: number;
     supply: number;
   } & StatsMap;
+  campaignId: string | undefined;
 }
 
 export function MovesSection(props: MovesSectionProps) {
-  const { stats } = props;
+  const { stats, campaignId } = props;
+  if (campaignId) useListenToCampaignCustomMoves(campaignId);
```

Comment:
Hooks should not be called conditionally - instead, useListenToCampaignCustomMoves should allow campaignId to be undefined.

This is because react relies on hooks being called in the same order on every single rerender. You can read more about that [here in the react docs](https://reactjs.org/docs/hooks-rules.html#explanation)

Reply:
I definitely need to read through the docs more carefully. Knowing just this one thing has cleared up a lot of confusion for issues I've had in the past. Thanks!